yarn-project/pxe/src/events/private_event_filter_validator.test.ts

```diff
@@ -0,0 +1,122 @@
+import { PrivateEventFilterValidator } from './private_event_filter_validator.js';
+
+describe('PrivateEventFilterValidator', () => {
+  let contractAddress: AztecAddress;
+  let lastKnownBlockNumber: BlockNumber;
+  let scope: AztecAddress;
+  let syncDataProvider: MockProxy<SyncDataProvider>;
+  let validator: PrivateEventFilterValidator;
+
+  beforeEach(async () => {
+    lastKnownBlockNumber = BlockNumber(42);
+    contractAddress = await AztecAddress.random();
+    scope = await AztecAddress.random();
+    syncDataProvider = mock<SyncDataProvider>();
+    syncDataProvider.getBlockNumber.mockResolvedValue(lastKnownBlockNumber);
+    validator = new PrivateEventFilterValidator(syncDataProvider);
+  });
+
+  it('rejects empty scope', async () => {
+    await expect(validator.validate({ contractAddress, fromBlock: lastKnownBlockNumber, scopes: [] })).rejects.toThrow(
+      /At least one scope is required to get private events/,
+    );
+  });
+
+  it('defaults to whole range', async () => {
+    const dataProviderFilter = await validator.validate({ contractAddress, scopes: [scope] });
+    expect(dataProviderFilter).toEqual({
+      contractAddress,
+      scopes: [scope],
+      fromBlock: BlockNumber(1),
+      toBlock: lastKnownBlockNumber + 1,
+      txHash: undefined,
+    });
+  });
+
+  it('toBlock defaults to lastKnownBlock + 1', async () => {
+    const dataProviderFilter = await validator.validate({
+      contractAddress,
+      scopes: [scope],
+      fromBlock: BlockNumber(1),
+    });
+    expect(dataProviderFilter).toEqual({
+      contractAddress,
+      scopes: [scope],
+      fromBlock: BlockNumber(1),
+      toBlock: BlockNumber(lastKnownBlockNumber + 1),
+    });
+  });
+
+  it('toBlock without fromBlock defaults to [0, toBlock)', async () => {
+    const dataProviderFilter = await validator.validate({
+      contractAddress,
+      scopes: [scope],
+      toBlock: BlockNumber(lastKnownBlockNumber + 1),
+    });
+    expect(dataProviderFilter).toEqual({
+      contractAddress,
+      scopes: [scope],
+      fromBlock: BlockNumber(1),
+      toBlock: BlockNumber(lastKnownBlockNumber + 1),
+    });
+  });
+
+  it('rejects fromBlock >= toBlock', async () => {
+    await expect(
+      validator.validate({
+        contractAddress,
+        scopes: [scope],
+        fromBlock: lastKnownBlockNumber,
+        toBlock: lastKnownBlockNumber,
+      }),
+    ).rejects.toThrow(/toBlock must be strictly greater than fromBlock/);
+
+    await expect(
+      validator.validate({
+        contractAddress,
+        scopes: [scope],
+        fromBlock: lastKnownBlockNumber,
+        toBlock: BlockNumber(lastKnownBlockNumber - 1),
+      }),
+    ).rejects.toThrow(/toBlock must be strictly greater than fromBlock/);
+  });
+
+  it('preserves txHash', async () => {
+    let dataProviderFilter = await validator.validate({
+      contractAddress,
+      scopes: [scope],
+      fromBlock: BlockNumber(1),
```

Comment:
```suggestion
      fromBlock: INITIAL_L2_BLOCK_NUM,
```

Would just use here the constant here and below given that we have it available.